### DISCUSSION TITLE
remote (classic) kernels, just like Meta Bento

### DISCUSSION THREAD

**Discussion by MRYingLEE** [jupyterlite/jupyterlite#1415] [Ideas]
Meta has an internal notebook platform named as Bento. ( https://engineering.fb.com/2024/06/10/data-infrastructure/serverless-jupyter-notebooks-bento-meta/). Bento is based on Jupyterlite.

![image](https://github.com/jupyterlite/jupyterlite/assets/11193534/f143c1f3-d658-4d08-a181-d3da1613e9f1)

Bento supports both local (browser) kernels and remote (classic) kernels.

Just suggest Jupyterlite also support remote (classic) kernels, such as AWS, GCP, and Azure based Jupyterlab services.

**Comment by jtpio** [comment]
Thanks @MRYingLEE for sharing this :+1: 

Indeed that would be great. There is some work happening upstream in JupyterLab to make the underlying Service Manager easier to customize, which should help JupyterLite with this use case at some point:
- https://github.com/jupyterlab/jupyterlab/issues/15329
- https://github.com/jupyterlab/lumino/issues/697

**Reply by psychemedia** [reply]
I would love to see support for remote kernels, eg published via [kernel gateway](https://jupyter-kernel-gateway.readthedocs.io/en/latest/), in JupyterLite. My use case is an ediucational setting where we can get away with running pyodide or xeus-python a lot of the time, handing over some "operating sysytem package" calls (eg to things like graphviz, pglite) via wasm packages wrapped using `anywidget` at other times. But sometimes, we need access to a "full fat" kernel, or a remote server with a bit more resource available to it.

**Reply by manics** [reply]
There was a nice demo of something similar (though not using JupyterLite) at last month's community meeting:
https://youtu.be/ildFScV6mZQ?t=660

**Comment by MRYingLEE** [comment]
Another related repo, Jupyter Kernel Client (through websocket) (https://github.com/datalayer/jupyter-kernel-client).

**Reply by psychemedia** [reply]
Ah, so that could be used via a custom IPyhton cell block magic  of the form `%%remote_kernel -u URL -t TOKEN`?
